index.js

```diff
@@ -138,47 +183,63 @@ function nowServing(line) {
 const line = []
 
 // ***** Scope & Closures - Question 1 *****
+let ticketNumber = 0
 
+function takeATicketNumber(line) {
+  ticketNumber += 1
+  line.push(ticketNumber)
 
+  return `Welcome. You are ticket number ${ticketNumber}`
+}
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
+
+console.log("%cScope & Closures - Question 1", "color: red")
+
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 1.`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
+
+function ticketNumberGeneratorFunc(newLine, takeATicketNumber) {
+    let ticketNumber = 0
+    ticketNumber += 1
+    newLine.push(ticketNumber)
+
+  return `Welcome. You are ticket number ${ticketNumber}`
+}
 // *** Uncomment the lines below to test
```

Comment:
This one's a tricky bonus one! Check this Scrimba out https://scrimba.com/scrim/cLgEmZUZ and see if you can figure this out after watching it and playing around with the examples. (In Scrimba, you can pause the video and play around with the code shown on the screen) (edited)